ProjectService_AddUserToProjectChanges

mekk1t opened the pull request:
Task:

- Add new properties

- Delete DbProjectManagerUser.cs

- Change implementation of the "Add user to project" -- WARNING! The task "Add user to project" was not even started, so i cannot make changes to that.

mekk1t committed:
Removed junk files

mekk1t committed:
DbProjectWorkerUser Model Update

Added new properties, deleted DbProjectManagerUser

KostinP commented on DigitalOffice/Services/ProjectService/Database/Entities/DbProject.cs:
```diff
@@ -13,7 +13,6 @@ public class DbProject
         public bool Deferred { get; set; }
         public bool IsActive { get; set; }
-        public ICollection<DbProjectManagerUser> ManagersUsersIds { get; set; }
         public ICollection<DbProjectWorkerUser> WorkersUsersIds { get; set; }
```
New migration?

mekk1t replied:
Yeah, will add.

My bad, working in a stressful environment

KostinP replied:
Write which team you will succeed with. I tried through "Add-Migration {Name} -Namespace {Namespace}", but it didn't work out for me.

mekk1t replied:
@KostinP you mean "command" not team, Pasha xD
I committed changes and wrote how-to there.
In my case, there were already some migrations in Database/Migrations/, so the standard command worked out.

KostinP replied:
I even stopped looking at what the translator gives me :D

mekk1t committed:
ManagersRemoved migration

Editor broke Caps and thats why migration name is in lower case
If that's critical, please address it and I'll rename it to PascalCase

Migration was created via Powershell.exe
while in project directory, typed:
dotnet ef migrations add managersremoved

Migrations were created in Database directory, the rest as usual.
@kostinp

KostinP commented:
Conflicts need to be resolved.

mekk1t committed:
Merge branch 'develop' into feature/ProjectService_AddUserToProjectChanges

mekk1t committed:
Mapper: Removed ManagersIds

Removed ManagersIds, because we do not have DbProjectManagerUser table anymore.

BUT! What if we still do have ManagersIds property, we just fill it with WorkerUserIds that have IsManager == true?

I think we should better filter it by code, via some method like "GetManagers" where dbContext.Projects.Include (p => p.WorkersUsersIds.Where (wId => wId.IsManager == true)
or something like that

mekk1t commented:
Errors building Tests project.
Huh, it did not come up somehow yesterday.

Will fix tests, by removing ManagerUserId usage and adding new properties to test and validate (if needed)

mekk1t committed:
Public bool properties + Tests Fixes

Removed ManagersIds from tests;

Updated DbProjectWorkerUser -- made bool properties public. Now migration recognized these properties and the table is updated. Screenshots will be attached

mekk1t commented:
![image](https://user-images.githubusercontent.com/61093487/89132257-96429280-d51b-11ea-87fb-7dd6ab46517b.png)
